spot-ml/src/main/scala/org/apache/spot/utilities/data/validation/InputSchema.scala

```diff
@@ -0,0 +1,48 @@
+package org.apache.spot.utilities.data.validation
+
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import scala.collection.mutable.ListBuffer
+
+
+/**
+  * Input schema routines
+  */
+object InputSchema {
+
+  // Response from validate will contains always at least 1 element. 1 message means no errors.
+  val ResponseDefaultSize = 1
+
+  /**
+    * Validate the incoming data schema matches the schema required for model creation and scoring.
+    *
+    * @param inSchema       incoming data frame
+    * @param expectedSchema schema expected by model training and scoring methods
+    * @return
+    */
+  def validate(inSchema: StructType, expectedSchema: StructType): Seq[String] = {
```

Comment:
to be explicit... this test should pass if there are extra columns  in the dataframe, not just the ones used by the model schema ?

this should be commented on and tested on